python/ray/data/_internal/planner/plan_read_op.py

```diff
@@ -72,14 +103,24 @@ def do_read(blocks: Iterable[ReadTask], _: TaskContext) -> Iterable[Block]:
             yield from read_task()
 
     # Create a MapTransformer for a read operator
-
-    # TODO(hchen): Currently, we apply the BlockOuputBuffer within the read tasks.
-    # We should remove that and use `_blocks_to_output_blocks` here.
     transform_fns = [
```

Comment:
add a comment explaining the logic for `transform_fns`. Why does `blocks_to_output_blocks` need to come before additional splitting?

Reply:
actually, I think they are redundant and should be exclusive. I'll fix it and add a comment.

Reply:
The original way was correct. I was dumb. The reason is because the read tasks produce small blocks. so we need to build large blocks first and do additional splitting.